packages/db-objects-file/lib/objects/objectsInMemServerRedis.js

```diff
@@ -455,18 +433,22 @@ class ObjectsInMemoryServer extends ObjectsInMemoryFileDB {
                     handler.sendBulk(responseId, JSON.stringify(obj));
                 } else {
                     // Handle request for File data
-                    this._readFile(id, name, (err, data, _mimeType) => {
-                        if (err || data === undefined || data === null) {
-                            handler && handler.sendNull(responseId);
-                        } else {
-                            if (!Buffer.isBuffer(data) && typeof data === 'object') {
-                                // if its an invalid object, stringify it and log warning
-                                data = JSON.stringify(data);
-                                this.log.warn(`${namespaceLog} Data of "${id}/${name}" has invalid structure at file data request: ${data}`);
-                            }
-                            handler && handler.sendBufBulk(responseId, Buffer.from(data));
-                        }
-                    });
+                    let data;
+                    try {
+                        data = this._readFile(id, name);
+                    } catch (err) {
+                        return void handler.sendNull(responseId);
+                    }
+                    if (!data || data.fileContent === undefined || data.fileContent === null) {
```

Comment:
can _readFile return null/undefined and not throw? Otherwise the `!data` is unnecessary.

Reply:
You are right, should only throw or return a struct. WIll change it